pkg/controller/utils/gcp/utils.go

```diff
@@ -147,22 +147,43 @@ func filterOutPermissions(permList, filterOutList []string) []string {
 // gcpClient creds have sufficient permissions to perform the actions.
 // Will return true/false indicating whether the permissions are sufficient.
 func CheckPermissionsAgainstPermissionList(gcpClient ccgcp.Client, permList []string, logger log.FieldLogger) (bool, error) {
+
+	if len(permList) == 0 {
+		return true, nil
+	}
+
 	projectName := gcpClient.GetProjectName()
 
 	filteredPermList := filterOutPermissions(permList, invalidPermsForProjectScopedPermissionsTesting)
 
-	permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
-		Permissions: filteredPermList,
-	}
-	permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
-	if err != nil {
-		return false, fmt.Errorf("error testing permissions: %v", err)
+	// Split list to only check 100 permissions at a time
+	responses := []*cloudresourcemanager.TestIamPermissionsResponse{}
+	done := false
+	index := 0
+	for !done {
+		upperLimit := len(filteredPermList)
```

Comment:
nit: I'd rather use @ironcladlou's [loop approach][1] here and just hard-code the expected chunking in the unit test.

```go
for _, testCase := range []struct{
  length int
  chunks []int
}{
  {length: 0},
  {length: 2, chunks: {2}},
  {length: 100, chunks: {100}},
  {length: 101, chunks: {100, 1}},
  {length: 230, chunks: {100, 100, 30}},
  {length: 300, chunks: {100, 100, 100}},
  {length: 379, chunks: {100, 100, 100, 79}},
} {
  t.Run(fmt.Sprintf("length=%d", testCase.length), ...)
}
```

[1]: https://github.com/openshift/cloud-credential-operator/pull/94#discussion_r307866955